salt/docker/init.sls

```diff
@@ -2,25 +2,35 @@ include:
   - repositories
   - flannel
 
-docker:
-  pkg.installed:
-    - install_recommends: False
-    - require:
-      - file: /etc/zypp/repos.d/containers.repo
-  service.running:
-    - enable: True
-    - watch:
-      - service: flannel
-      - file: /etc/sysconfig/docker
-    - require:
-      - pkg: docker
+{% set proxy_http = salt['pillar.get']('proxy:http', '') %}
+{% if proxy_http|length > 0 %}
+/etc/systemd/system/docker.service.d/http-proxy.conf:
+  file.managed:
+    - content: |
+      [Service]
+      Environment="HTTP_PROXY={{ proxy_http }}"
+    - watch_in:
+      - service: docker
+{% endif %}
+
+{% set proxy_https = salt['pillar.get']('proxy:https', '') %}
+{% if proxy_https|length > 0 %}
+/etc/systemd/system/docker.service.d/https-proxy.conf:
+  file.managed:
+    - content: |
+      [Service]
+      Environment="HTTPS_PROXY={{ proxy_https }}"
+    - watch_in:
+      - service: docker
+{% endif %}
```

Comment:
These two should be updated to remove the proxy when the pillar is removed. Just removing the if|length should do it, unless systemd doesn't like `Environment="HTTPS_PROXY=`. At that point, both files can also be combined into one.